server/etcdserver/server.go

```diff
@@ -1771,50 +1771,56 @@ func (s *EtcdServer) apply(
 	s.lg.Debug("Applying entries", zap.Int("num-entries", len(es)))
 	for i := range es {
 		e := es[i]
-		index := s.consistIndex.ConsistentIndex()
-		s.lg.Debug("Applying entry",
-			zap.Uint64("consistent-index", index),
-			zap.Uint64("entry-index", e.Index),
-			zap.Uint64("entry-term", e.Term),
-			zap.Stringer("entry-type", e.Type))
-
-		// We need to toApply all WAL entries on top of v2store
-		// and only 'unapplied' (e.Index>backend.ConsistentIndex) on the backend.
-		shouldApplyV3 := membership.ApplyV2storeOnly
-		if e.Index > index {
-			shouldApplyV3 = membership.ApplyBoth
-			// set the consistent index of current executing entry
-			s.consistIndex.SetConsistentApplyingIndex(e.Index, e.Term)
-		}
-		switch e.Type {
-		case raftpb.EntryNormal:
-			// gofail: var beforeApplyOneEntryNormal struct{}
-			s.applyEntryNormal(&e, shouldApplyV3)
-			s.setAppliedIndex(e.Index)
-			s.setTerm(e.Term)
-
-		case raftpb.EntryConfChange:
-			// gofail: var beforeApplyOneConfChange struct{}
-			var cc raftpb.ConfChange
-			pbutil.MustUnmarshal(&cc, e.Data)
-			removedSelf, err := s.applyConfChange(cc, confState, shouldApplyV3)
-			s.setAppliedIndex(e.Index)
-			s.setTerm(e.Term)
-			shouldStop = shouldStop || removedSelf
-			s.w.Trigger(cc.ID, &confChangeResponse{s.cluster.Members(), raftAdvancedC, err})
-
-		default:
-			lg := s.Logger()
-			lg.Panic(
-				"unknown entry type; must be either EntryNormal or EntryConfChange",
-				zap.String("type", e.Type.String()),
-			)
-		}
+		removedSelf := s.applyEntry(&e, confState, raftAdvancedC)
+		shouldStop = shouldStop || removedSelf
 		appliedi, appliedt = e.Index, e.Term
 	}
 	return appliedt, appliedi, shouldStop
 }
 
+func (s *EtcdServer) applyEntry(e *raftpb.Entry, confState *raftpb.ConfState, raftAdvancedC <-chan struct{}) (shouldStop bool) {
```

Comment:
IIUC, `shouldStop` should be `removedSelf`

Reply:
done